### PR TITLE
Add hash check to the upgrade script

### PR DESCRIPTION
**Pull request by galal-hussein**
https://github.com/rancher/k3s/issues/1408

### CODE REVIEW

File: scripts/upgrade.sh [hunk 1]
```diff
@@ -47,7 +47,25 @@ kill_k3s_process() {
...
+check_hash(){
+    sha_cmd="sha256sum"
+
+    if [ ! -x "$(command -v $sha_cmd)" ]; then
+    sha_cmd="shasum -a 256"
+    fi
+
+    if [ -x "$(command -v $sha_cmd)" ]; then
+
+    K3S_RELEASE_CHECKSUM=$(echo $K3S_RELEASE_CHECKSUM | sed "s/sha256sum.txt/sha256sum-amd64.txt/g")
```
Comment (dweomer) [inline scripts/upgrade.sh]: This is because the lack of `-amd64` suffix on builds for that arch?

File: Dockerfile [hunk 2]
```diff
@@ -3,9 +3,10 @@ FROM alpine:3.10
 ARG ARCH=
 
 ENV K3S_RELEASE https://github.com/rancher/k3s/releases/download/${DRONE_TAG}/k3s${ARCH}
+ENV K3S_RELEASE_CHECKSUM https://github.com/rancher/k3s/releases/download/${DRONE_TAG}/sha256sum${ARCH}.txt
 
-RUN wget -O /bin/k3s ${K3S_RELEASE}
-RUN chmod +x /bin/k3s
+RUN wget -O /opt/k3s ${K3S_RELEASE}
+RUN chmod +x /opt/k3s
```
Comment (dweomer) [inline Dockerfile]: Why the change to `/opt`? Assuming there is a good reason for the prefix change, shouldn't this be `/opt/bin/k3s`?

File: scripts/upgrade.sh (hunk 1 above)
Comment (galal-hussein) [reply scripts/upgrade.sh]: yes

File: Dockerfile (hunk 2 above)
Comment (galal-hussein) [reply Dockerfile]: no particular reason I just wanted the binary to be placed in /opt to be checked first against the hash and then copied to the right location
